### PR TITLE
- Migrate syntax to support swift 3

### PR DESCRIPTION
**Pull request by kevin-chen-bcgdv**
I've run the unit tests, all test cases pass

### CODE REVIEW

**Comment by kevin-chen-bcgdv** [comment]
It looks like the Travis hasn't setup for swift 3

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "Carthage/Checkouts/CryptoSwift"]
 	path = Carthage/Checkouts/CryptoSwift
-	url = https://github.com/krzyzanowskim/CryptoSwift.git
+	url = https://github.com/xmkevinchen/CryptoSwift.git
```
Comment (kylef) [inline .gitmodules]: I don't want to move to a third party fork, but rather stick to the original CryptoSwift.

File: JSONWebToken.podspec [hunk 2]
```diff
@@ -5,12 +5,12 @@ Pod::Spec.new do |spec|
...
   spec.author = { 'Kyle Fuller' => 'kyle@fuller.li' }
-  spec.source = { :git => 'https://github.com/kylef/JSONWebToken.swift.git', :tag => "#{spec.version}" }
+  spec.source = { :git => 'https://github.com/kevin-chen-bcgdv/JSONWebToken.swift.git', :branch => "swift-3" }
   spec.source_files = 'Sources/*.swift'
   spec.ios.deployment_target = '8.0'
   spec.osx.deployment_target = '10.9'
   spec.tvos.deployment_target = '9.0'
   spec.requires_arc = true
-  spec.dependency 'CryptoSwift', '~> 0.4.0'
+  spec.dependency 'CryptoSwift'
```
Comment (kylef) [inline JSONWebToken.podspec]: This basically means that people can use unsupported versions of CryptoSwift with JSONWebToken. Along with if there are future incompatible releases of CryptoSwift break JSONWebToken for users in the future.

We should always pin to the API compatible version.

File: JWT.xcodeproj/project.xcworkspace/xcshareddata/JWT.xcscmblueprint [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (kylef) [inline JWT.xcodeproj/project.xcworkspace/xcshareddata/JWT.xcscmblueprint]: This file is local Xcode state and should not be committed to the repository.

**Comment by kylef** [comment]
Thanks for the pull request Kevin. I've gone ahead and added Swift 3 support in JSONWebToken 2.0.0. There are a couple of reasons why I haven't merged your pull request that you can see above. Along with the Travis CI build failures.

Regards,

Kyle

**Pull request closed without merging**